### ISSUE TITLE
Hello example failing to build

### ISSUE THREAD

**Issue by jcbritobr** [gotham-rs/gotham#400]
The example https://github.com/gotham-rs/gotham/tree/master/examples/hello_world fails to build. Seems there is a wrong module being imported, or api had changes and example wasn't updated

 
assert_eq!(response.status(), StatusCode::Ok);
                                ^^^^^^^^^^ use of undeclared type or module `StatusCode`

**Comment by pksunkara** [comment]
Hey, how are you building it? We had re-exported `hyper` in gotham.

**Comment by jcbritobr** [comment]
> Hey, how are you building it? We had re-exported `hyper` in gotham.

Hello. Im just compiling with cargo. 'cargo build'

There is no additional configuration in cargo.toml

**Comment by kahlil29** [comment]
I got hit by this a couple of times when trying out some examples. 
I think the problem boils down to the fact that in the `Cargo.toml` of the examples, the `local` version of `gotham` is used a dependency, but I had pointed to the latest version in my `Cargo.toml`

**Comment by pksunkara** [comment]
Yup, you are not supposed to modify the `Cargo.toml` from examples. But I think @colinbankier should do a pre-release soon and finalize the roadmap for 5.0 so we can actually push 5.0

**Comment by kahlil29** [comment]
I would think that most people (atleast this is what I do, usually) would compile and run example code and then copy paste snippets or the entire thing into a project in which they want to use the code/tool. 
So imo, there should be either a note or gotcha in the documentation along with something that would guide or help people using the library to use the example code/concept in their code. I'm willing to write this/open a PR for this if someone can help/review/guide me 😄

**Comment by jcbritobr** [comment]
If we can't run a simple code snnipet, with simple cargo confuguration, I
think we have an issue in documentation.

Em seg, 24 de fev de 2020 00:02, Kahlil Abreu <notifications@github.com>
escreveu:

**Comment by blueskyjunkie** [comment]
@kahlil29 Could you post your proposed solution here? So at least in absence of documentation those (like me) searching for a solution to this are somewhat likely to find this issue and be able to use your solution.

**Comment by kahlil29** [comment]
@blueskyjunkie 
Unfortunately I cannot remember vividly, but could you try adding `hyper = "0.12.0"` to your `Cargo.toml` and see if it works? 
If I remember correctly, after some fiddling around with the versions, it worked.

**Comment by blueskyjunkie** [comment]
@kahlil29 Gotcha. It seems you have to make sure the versions of hyper and http you import into your project are exactly the same as those used by gotham itself.

Presumably there's a Cargo way to do this, but in the Intellij IDEA IDE project I just looked at the "External libraries" drop down tree in the project structure to find which versions of http and hyper are imported automatically by gotham.

![image](https://user-images.githubusercontent.com/11006658/79228990-518dd100-7e30-11ea-9c96-04b70fa16cb4.png)

and then made sure those versions are imported in my `Cargo.toml` (I'm importing them as test dependencies for now):

![image](https://user-images.githubusercontent.com/11006658/79229106-7f731580-7e30-11ea-9be7-41f652dac394.png)

Definitely a documentation issue at the moment, but also comprises a maintenance issue because you have to manually track the http/hyper versions in your own project. I think the fix is for gotham to publically export it's hyper/http dependencies for transparent use by a dependent project like hyper does for it's http dependencies.

**Comment by sinistersnare** [comment]
Yes, what is the status of 0.5. There are significant differences between what the examples have and what is needed to make it work with 0.4. When can a 0.5 be expected?

**Comment by colinbankier** [comment]
> Yes, what is the status of 0.5. There are significant differences between what the examples have and what is needed to make it work with 0.4. When can a 0.5 be expected?

@nyarly - are you able to release a gotham 0.5.0-rc.0? Just responding here in case you don't see the chat in gitter about it.

**Comment by msrd0** [comment]
Version `0.5.0` has been released so the issues mentioned above should've been fixed, but examples on the master branch will always track master, so I'm not really sure how you propose this to not happen again in the future @kahlil29?

**Comment by nyarly** [comment]
I'd think that the thing to do really is a documentation change - add comments to the examples themselves, or their READMEs or the README in the top level examples directory. Something to the effect that if you're going to C&P, be aware that versions are still very significant, and you'll want to work from the tag matching the version of Gotham that you're using.

As I'm formulating that response, I think there's a overarching issue with a unified docs strategy.
